### PR TITLE
Fix key clash in search query rules (bug 998070)

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

**Comment by robhudson** [comment]
The bug was to add a match query on name but it turned out we already had one but it wasn't showing up in the query b/c of the dictionary key clash, so this bug ended up being easy.

File: mkt/search/filters.py [hunk 1]
```diff
@@ -34,18 +34,20 @@ def filter_queryset(self, request, queryset, view):
...
 
         # Only add fuzzy queries if q is a single word. It doesn't make sense
         # to do a fuzzy query for multi-word queries.
         if ' ' not in q:
-            rules[query.Fuzzy] = {'value': q, 'boost': 2, 'prefix_length': 1}
+            rules.append(
+                (query.Fuzzy, {'value': q, 'boost': 2, 'prefix_length': 1}))
 
-        for k, v in rules.iteritems():
+        for k, v in rules:
```
Comment (chuckharmston) [inline mkt/search/filters.py]: Ha, good catch. I would have missed this.

File: mkt/search/filters.py [hunk 2]
```diff
@@ -34,18 +34,20 @@ def filter_queryset(self, request, queryset, view):
...
-        rules = {
-            query.Match: {'query': q, 'boost': 3, 'analyzer': 'standard'},
-            query.Match: {'query': q, 'boost': 4, 'type': 'phrase', 'slop': 1},
-            query.Prefix: {'value': q, 'boost': 1.5}
-        }
+        rules = [
+            (query.Match, {'query': q, 'boost': 3, 'analyzer': 'standard'}),
+            (query.Match, {'query': q, 'boost': 4, 'type': 'phrase',
+                           'slop': 1}),
+            (query.Prefix, {'value': q, 'boost': 1.5}),
```
Comment (chuckharmston) [inline mkt/search/filters.py]: I guess this is legit now that Bastacorp has filed for chapter 11.

**Comment by chuckharmston** [comment]
r+

**Pull request merged**